share/scan/compliance.go

```diff
@@ -3207,89 +3222,43 @@ func LoadConfig(primeConfig string, params *UpdateConfigParams, updateCompliance
 		} else {
 			updateImageBenchWithPrimeConfig(primeConfig, params)
 		}
-		return
-	}
-
-	watcher, err := fsnotify.NewWatcher()
-	if err != nil {
-		log.WithFields(log.Fields{"error": err}).Error("Init watcher error ")
-		return
-	}
-
-	defer watcher.Close()
-	done := make(chan bool)
-
-	if _, err := os.Stat(filepath.Dir(primeConfig)); os.IsNotExist(err) {
-		log.WithFields(log.Fields{"directory": filepath.Dir(primeConfig)}).Info("Directory does not exist, skipping watcher setup")
-		return
-	} else {
-		err = watcher.Add(filepath.Dir(primeConfig))
-		if err != nil {
-			log.WithFields(log.Fields{"error": err}).Error("Failed to add directory to watcher")
-			return
-		}
 	}
-
-	go func() {
-		defer close(done)
-		for {
-			select {
-			case event, ok := <-watcher.Events:
-				if !ok {
-					return
-				}
-				switch {
-				case event.Op&fsnotify.Create == fsnotify.Create && event.Name == primeConfig:
-					if updateCompliance {
-						updateComplianceWithPrimeConfig(primeConfig, params)
-					} else {
-						updateImageBenchWithPrimeConfig(primeConfig, params)
-					}
-					return
-				case event.Op&fsnotify.Write == fsnotify.Write && event.Name == primeConfig:
-					if updateCompliance {
-						updateComplianceWithPrimeConfig(primeConfig, params)
-					} else {
-						updateImageBenchWithPrimeConfig(primeConfig, params)
-					}
-					return
-				}
-			case err, ok := <-watcher.Errors:
-				if !ok && err == nil {
-					return
-				}
-				log.WithFields(log.Fields{"error": err, "ok": ok}).Error("LoadPrimeConfig error: ")
-				return
-			}
-		}
-	}()
-	<-done
 }
 
 func UpdateComplianceConfigs() {
-	// update prime cis configs
-	primeCISConfig := fmt.Sprintf("%s%s.yaml", primeConfigPrefix, cisVersion)
-
-	// update prime cis docker configs
-	primeDockerConfig := fmt.Sprintf("%s%s.yaml", primeConfigPrefix, "cis-docker")
+	ReadPrimeConfig = false
```

Comment:
`UpdateComplianceConfigs()` is called in a [go routine](https://github.com/neuvector/neuvector/blob/main/controller/controller.go#L745)

And `updateComplianceWithPrimeConfig()` and `updateImageBenchWithPrimeConfig()` are called in different go routines, e.g., [this](https://github.com/neuvector/neuvector/blob/main/share/scan/compliance.go#L3233) and [this](https://github.com/neuvector/neuvector/blob/main/share/scan/compliance.go#L3291)

You probably have to protect the flag using sync.Mutex.